core/src/main/java/io/grpc/internal/ProxyDetectorImpl.java

```diff
@@ -258,12 +260,14 @@ private ProxyParameters detectProxy(InetSocketAddress targetAddr) throws IOExcep
     }
 
     if (auth == null) {
-      return new ProxyParameters(resolvedProxyAddr, null, null);
+      return ProxyParameters.forAddress(resolvedProxyAddr).build();
     }
 
-    // TODO(spencerfang): users of ProxyParameters should clear the password when done
-    return new ProxyParameters(
-        resolvedProxyAddr, auth.getUserName(), new String(auth.getPassword()));
+    return ProxyParameters
+        .forAddress(resolvedProxyAddr)
+        .username(auth.getUserName())
+        .password(new String(auth.getPassword()))
```

Comment:
hmmm, do you know why user name can be null, but not password?

Reply:
fixed